### PR TITLE
fix(sdcm/send_email.py): fix cut_report_data

### PR DESCRIPTION
**Pull request by dkropachev**
https://github.com/scylladb/scylla-cluster-tests/issues/1813

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by fruch** [comment]
@dkropachev can we not put a 21.8 MB json into the git repo ?
I would rather configure a smaller size for the test, or generate the file inside the test.

**Comment by dkropachev** [comment]
> @dkropachev can we not put a 21.8 MB json into the git repo ?
> I would rather configure a smaller size for the test, or generate the file inside the test.



> @dkropachev can we not put a 21.8 MB json into the git repo ?
> I would rather configure a smaller size for the test, or generate the file inside the test.

zipped it. to 700kb

File: sdcm/send_email.py [hunk 1]
```diff
@@ -63,7 +63,8 @@ def prepare_email(self, subject, content, recipients, html=True, files=()):  # p
         msg = MIMEMultipart()
         msg['subject'] = subject
         msg['from'] = self.sender
-        msg['to'] = ','.join(recipients)
+        if recipients:
```
Comment (bentsi) [inline sdcm/send_email.py]: why not checking all  msg['subject'], msg['from']??
I think we shouldn't check it here. We shouldn't run the method when recipients is empty

File: sdcm/send_email.py [hunk 2]
```diff
@@ -180,6 +181,9 @@ def send_report(self, results):
         if email is None:
             self.log.error("Failed to prepare email", exc_info=True)
             return
+        if not self.email_recipients:
```
Comment (bentsi) [inline sdcm/send_email.py]: move this code to the beginning of the method, and set log level as warning

File: sdcm/send_email.py (hunk 1 above)
Comment (dkropachev) [reply sdcm/send_email.py]: it is done to be able to test whole report cycle except actual email sending, by setting recipient to None.

File: sdcm/send_email.py (hunk 2 above)
Comment (dkropachev) [reply sdcm/send_email.py]: It's purpose is to be able to test this method, skipping sending part. So it is perfect place for it

File: sdcm/send_email.py [hunk 3]
```diff
@@ -228,7 +233,7 @@ class LongevityEmailReporter(BaseEmailReporter):
               'username', 'nodes', 'events_summary']
 
     def cut_report_data(self, report_data, attachments_data, reason):
-        if ['test_status'] in attachments_data and len(attachments_data['test_status']) > 2 and len(
+        if attachments_data and 'test_status' in attachments_data and len(attachments_data['test_status']) > 2 and len(
```
Comment (aleksbykov) [inline sdcm/send_email.py]: Why leave >2, ['test_status'] has length 2 always

Comment (dkropachev) [reply sdcm/send_email.py]: Thanks Alex, second time you point it out to me ]

File: sdcm/utils/thread.py [hunk 4]
```diff
@@ -107,7 +109,7 @@ def format_error(exc):
         :param exc: the exception
         :return: string to add to the event
         """
-        #  pylint: disable=no-member
+        #  pylint: disable=all
```
Comment (fruch) [inline sdcm/utils/thread.py]: why did you disabled those checked ?

Comment (dkropachev) [reply sdcm/utils/thread.py]: 19:45:26  ************* Module multiple_dc_test
19:45:26  multiple_dc_test.py:1:0: R0801: Similar lines in 2 files
19:45:26  ==sdcm.stress_thread:167
19:45:26  ==sdcm.utils.thread:112
19:45:26                  if hasattr(exc, 'result') and exc.result.failed:
19:45:26                      stderr = exc.result.stderr
19:45:26                      if len(stderr) > 100:
19:45:26                          stderr = stderr[:100]
19:45:26                      errors_str = f'Stress command completed with bad status {exc.result.exited}: {stderr}'
19:45:26                  else:
19:45:26                      errors_str = f'Stress command execution failed with: {str(exc)}' (duplicate-code)
19:45:26  
19:45:26  --------------------------------------------------------------------
19:45:26  Your code has been rated at 10.00/10 (previous run: 10.00/10, -0.00)

Unfortunately there is some kind of problem around disabling duplicate-code, disable all handlers is only thing that helped.

File: sdcm/send_email.py (hunk 2 above)
Comment (dkropachev) [reply sdcm/send_email.py]: FIxed.

File: sdcm/utils/thread.py (hunk 4 above)
Comment (fruch) [reply sdcm/utils/thread.py]: I agree it's weird it's jumps out in other PRs in not the one initiated it (or other PRs that pass just fine), and that duplicate-code can't be disabled, only per file or global (it's a bug in pylint)

but doing `#  pylint: disable=all` beats the whole point of using pylint... 

please remove all the duplicate-code comment that doesn't actually work.
just change the code a bit, or refactor into a function. (I know it's my fault, that we have this duplicate code, if you want I can do that in a separate PR)

Comment (fruch) [reply sdcm/utils/thread.py]: this should solve the duplicate-code
https://github.com/scylladb/scylla-cluster-tests/pull/1830

Comment (dkropachev) [reply sdcm/utils/thread.py]: Ok, i fall it back.

**Pull request merged**